pkg/state/create.go

```diff
@@ -196,9 +196,14 @@ func (c *StateCreator) ParseAndLoad(content []byte, baseDir, file string, envNam
 }
 
 func (c *StateCreator) loadBases(envValues, overrodeEnv *environment.Environment, st *HelmState, baseDir string) (*HelmState, error) {
+	var newOverrodeEnv *environment.Environment
+	if overrodeEnv != nil {
+		overrodeEnvCopier := overrodeEnv.DeepCopy()
+		newOverrodeEnv = &overrodeEnvCopier
+	}
 	layers := []*HelmState{}
 	for _, b := range st.Bases {
-		base, err := c.LoadFile(envValues, overrodeEnv, baseDir, b, false)
+		base, err := c.LoadFile(envValues, newOverrodeEnv, baseDir, b, false)
```

Comment:
@yxxhero Thanks for the fix! Just wondering, but why this matters only for v1 mode?

Reply:
@mumoshu because of this PR: https://github.com/helmfile/helmfile/pull/705